### DISCUSSION TITLE
New OpenAI Assistants / Assistants API

### DISCUSSION THREAD

**Discussion by bbence84** [microsoft/autogen#575] [General]
I am trying to understand if the new OpenAI Assistants API is different conceptually or even technically from Autogen? Is it really the same idea and almost same approach, just packaged in a more accessible way? :)

**Comment by MrXandbadas** [comment]
openAI have integrated specific API calls for Agents. Which will help streamline a project like this, although it would probably require a rebase as its a whole different set of api calls. Instead of the /v1/completions its /v1/assistants

```
curl https://api.openai.com/v1/assistants/asst_abc123 \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $OPENAI_API_KEY" \
  -H "OpenAI-Beta: assistants=v1" \
  -d '{
      "instructions": "You are an HR bot, and you have access to files to answer employee questions about company policies. Always  response with info from either of the files.",
      "tools": [{"type": "retrieval"}],
      "model": "gpt-4",
      "file_ids": ["file-abc123", "file-abc456"]
    }'
```

Still watching the keynote xD

**Reply by goodcojoe** [reply]
Does anybody know if Autogen will support the new ChatGPT assistants? It seems like this should be incredibly high priority? The ability to make different flavours of agents would be huge

**Reply by gagb** [reply]
Yes, we are aware and put this issue on the roadmap (cc @sonichi)

**Reply by sonichi** [reply]
I want to get it done within a week, sooner the better.

**Reply by sonichi** [reply]
#602 #616

**Comment by bbence84** [comment]
I was more interested in e.g. whether OpenAI took heavy inspiration from the design of existing frameworks like Autogen or whether there's something that is fundamentally different that will only be possible using the OpenAI Agents framework. I mean, at first look, it seems that you are able to do everything already in Autogen that is currently exposed. But I would like to hear your thoughts. :)

**Reply by MrXandbadas** [reply]
I think frameworks like Autogen, Chatdev have always been in the background of their planning and thinking, you kinda can't shake how good these systems are at certain things given the right tools or structure to operate.
I can only hope that they took the rich featuresets developed and explored in the community forward with them and also expanded upon them with new features and implementations.

Let's look at "creating GPTs" as a simple demonstration of this thought.
An agent that has one goal, help the user setup an agentic workflow for reuse. Such a simple yet effective way to break the code barrier for agentic setup. 

Now let's humour the thought of making this agent capable of setting up multiple agentic workflows that can co-operate to perform duties. This is well within the scope of possibilities. 

Side note - GitHub is integrating AI agentic workflows across the board. Agents will and have already revolutionised the way we interact with web 2.0 and this is just the beginning.

All things noted are almost identical to the agentic flows we setup here in Autogen. They would all work almost identically as it's a core foundational idea with programmatic data handling out the wazooo to achieve it.

I can't even imagine what the future will bring for the agentic landscape in commercial form and in opensource. Especially when we give agents tooling like never before.

It's akin to when we learnt how to use spears, axes and fire.

Edit:
GodDamn thats fast:[OpenAI - Agent Swarm](https://github.com/daveshap/OpenAI_Agent_Swarm)

**Reply by goodcojoe** [reply]
From my understanding, OpenAI look to have introduced GPTs, but left some detail that would be very easy for them to implement out of the mix for this first iteration.

Namely. The ability for GPTs to call external APIs and other GPTs from within the native ChatGPT interface. This can be achieved by externally executed frameworks (like Autogen)... this must have been intentional.

I presume OpenAI are thinking long and hard about introducing external API calls and GPT <-> GPT conversations, I strongly doubt it's a lack of engineering prowess or time..

**Comment by IANTHEREAL** [comment]
Let's look forward. Next it may involves defining a standardized agent protocol that allows Autogen and OpenAI's GPTs to call each other. There's also significant potential in this opportunity for Autogen, as AutoGen can ensure that we never locked in just OpenAI but can work with various other Language Model (LLM) providers as well. 

The interface of OpenAI Assistant is simple and user-friendly. However, I'm currently not certain if it can address most complex issues. In this regard, my intuition tells me that domain experts might be needed for process orchestration, and this is where Autogen may find an opportunity.

Therefore, I believe Autogen should focus on strengthening its infrastructure capabilities in this regard.

**Reply by ProdManager** [reply]
Autogen being able to use other LLMs is the diferentiator.
OPenAI howver, just made it very easy to enter the agent world with a simple framework + no code agents via GPTs. They are thinking beyond the developer to the user and use case. 

So an autogen GPTs version?

**Reply by Knucklessg1** [reply]
Autogen can excel in agent orchestration as well as agent hierarchies. That's the next logical evolution in this. How we organize different LLM provider agents into autogen's grouping framework is what will provide value.

**Comment by louis030195** [comment]
https://github.com/stellar-amenities/assistants

**Reply by sonichi** [reply]
Thanks for sharing.
cc @gagb @IANTHEREAL
